src/devrules/core/git_service.py

```
@@ -448,3 +417,85 @@ def get_current_repo_name() -> str:
+
+def commit(message: str, config: Config):
+    """Commit staged changes."""
+    options = []
+    if config.commit.gpg_sign:
+        options.append("-S")
+    if config.commit.allow_hook_bypass:
+        options.append("-n")
+    options.append("-m")
+    options.append(message)
+    try:
+        subprocess.run(["git", "commit", *options], check=True)
```

Ruff (0.14.13)

`subprocess` call: check for execution of untrusted input

(S603)